test/util/fake-timers-test.js

```diff
@@ -102,72 +103,84 @@ describe("fakeTimers.clock", function() {
             this.clock = fakeTimers.clock.create();
         });
 
-        it("returns numeric id or object with numeric id", function() {
-            var result = this.clock.setImmediate(function() {
-                return;
-            });
+        if (typeof setImmediate === "function") {
```

Comment:
I wish we had a rule to detect this. This should be using `this.skip()` in the `before` instead of this.

Reply:
It shouldn't be that difficult to write an ESLint rule that detects when an `if` statement wraps an `it` invocation

Reply:
We had our reasons 🤷‍♂  https://github.com/sinonjs/sinon/pull/2086#discussion_r327539951